src/components/vm/disks/diskAdd.jsx

```diff
@@ -504,11 +505,38 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                // Limit process time to 2 seconds and ram to 100 MB
+                cockpit.spawn(["/bin/bash", "-c", "ulimit -t 2 -v 100000; qemu-img info --output json " + value], { err: "message" })
```

Comment:
Please add `-e` bash option, to ensure that the ulimit call succeeds.

But moreover, `value` must be shell-escaped here, to work with quotes and spaces. (Please add a test for a file name containing these). I'm not sure ATM how to do that in a sensible manner, it's rather awkward to have to go through a shell string. Perhaps you can avoid this by sending the file name via `.input()` and change the shell command to read it from stdin?

Reply:
Right, but you can use `xargs` to read a file name from stdin and call a program with it:

```
$ touch '/tmp/evil file"na'"'"'me'
$ stat -c %n /tmp/evil*
/tmp/evil file"na'me
$ echo -n '/tmp/evil file"na'"'"'me' | xargs -0n1 qemu-img info --output json
{
    "virtual-size": 0,
    "filename": "/tmp/evil file\"na'me",
    "format": "raw",
    "actual-size": 0,
    "dirty-flag": false
}
```

I know this is ugly, but as soon as you are in shell land, one has to resort to such measures. A better solution would be to teach the bridge to set ulimits on its own (via some cockpit.spawn() options), but I think let's not block on that.

Reply:
Ah, that's the magic I was missing. Thanks!